apps/web/src/actions/user-profile-mod.ts

```diff
@@ -12,40 +12,192 @@ import { revalidatePath } from "next/cache";
 // TODO: Add skill updating
 export const modifyRegistrationData = authenticatedAction(
 	z.object({
```

Comment:
nit and non-blocking: As stated before, common types should be in their own types and also use the types of the tables instead of manually adding them.